src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java

```diff
@@ -28,82 +27,147 @@ public void setUp() throws Exception {
-        Assert.assertEquals("$^{31}$", hm.get("review"));
+            Assert.assertEquals(1, risEntries.size());
+            Assert.assertEquals(1, bibEntries.size());
 
-        hm.put("title", "/sub $Hello/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_{\\$Hello}$", hm.get("title"));
+            BibtexEntryAssert.assertEquals(bibEntries, risEntries);
+        }
     }
 
-    @Test
-    public void testImportEntries() throws IOException {
+    @Test // Test for jabref/issues/296
+    public void testImportEntriesForTagT2Override() throws IOException {
         RisImporter importer = new RisImporter();
 
-        try (InputStream stream = RISImporterTest.class
-                .getResourceAsStream("RisImporterTest1.ris")) {
+        try (InputStream stream = RISImporterTest.class.getResourceAsStream("RisImporterTest3.ris")) {
             List<BibEntry> entries = importer.importEntries(stream, new OutputPrinterToNull());
-            Assert.assertEquals(1, entries.size());
             BibEntry entry = entries.get(0);
-            Assert.assertEquals("Editorial: Open Source and Empirical Software Engineering", entry
-                    .getField("title"));
-            Assert.assertEquals(
-                    "Harrison, Warren",
-                    entry.getField("author"));
-
-            Assert.assertEquals("article", entry.getType());
-            Assert.assertEquals("Empirical Software Engineering", entry.getField("journal"));
-            Assert.assertEquals("2001", entry.getField("year"));
-            Assert.assertEquals("6", entry.getField("volume"));
-            Assert.assertEquals("3", entry.getField("number"));
-            Assert.assertEquals("193--194", entry.getField("pages"));
-            Assert.assertEquals("#sep#", entry.getField("month"));
+            Assert.assertEquals(1, entries.size());
+
+            Assert.assertEquals("Dermoscopy Image Analysis", entry.getField("booktitle"));
+
+        }
+    }
+
+    @Test //Test for https://github.com/JabRef/jabref/pull/505#issuecomment-172629932
+    public void testImportEntriesWithMainFile() throws IOException {
+        RisImporter risImporter = new RisImporter();
+        BibtexImporter bibImporter = new BibtexImporter();
+
+        try (InputStream risStream = RISImporterTest.class.getResourceAsStream("RisImporterTest4.ris");
+                InputStream bibStream = BibtexImporter.class
+                        .getResourceAsStream("RisImporterTest4_comparefile.bib");) {
+
+            List<BibEntry> risEntries = risImporter.importEntries(risStream, new OutputPrinterToNull());
+            List<BibEntry> bibEntries = bibImporter.importEntries(bibStream, new OutputPrinterToNull());
+
+            Assert.assertEquals(risEntries.size(), bibEntries.size());
+
+            int referenceCount = 0;
+            for (int i = 0; i < risEntries.size(); i++) {
+                for (int j = 0; j < bibEntries.size(); j++) {
```

Comment:
Wait until #982 is merged in, then a simpler BibtexEntryAssert.assertEquals() should work.